lib/index.js

```diff
@@ -299,6 +299,54 @@ function parse(options) {
     }
 }
 
+/**
+ * parseSource
+ *
+ * @param {Object} options        Overwrite default options.
+ */
+function parseSource(source, options) {
+    options = _.defaults({}, options, defaults);
+
+    // extend with custom functions
+    app.filters = _.defaults({}, options.filters, app.filters);
+    app.languages = _.defaults({}, options.languages, app.languages);
+    app.parsers = _.defaults({}, options.parsers, app.parsers);
+    app.workers = _.defaults({}, options.workers, app.workers);
+    app.hooks = _.defaults({}, options.hooks, app.hooks);
+    // options
+    app.options = options;
+
+    // generator
+    app.generator = _.defaults({}, app.generator, defaultGenerator);
+
+    // packageInfos
+    app.packageInfos = _.defaults({}, app.packageInfos, defaultPackageInfos);
+
+    try {
+        // Log version information
+        var filename = path.join(__dirname, '../', './package.json');
+        var packageJson = JSON.parse(fs.readFileSync(filename, 'utf8'));
+        app.log.verbose('apidoc-generator name: ' + app.generator.name);
+        app.log.verbose('apidoc-generator version: ' + app.generator.version);
+        app.log.verbose('apidoc-core version: ' + packageJson.version);
+        app.log.verbose('apidoc-spec version: ' + getSpecificationVersion());
+
+        new PluginLoader(app);
+        var parser = new Parser(app);
+        var worker = new Worker(app);
+        var filter = new Filter(app);
+
+        // Make them available for plugins
+        app.parser = parser;
+        app.worker = worker;
+        app.filter = filter;
```

Comment:
Until here your whole code is the same as in `function parse(options) { ` So you should put this parts in an extra function or combine both functions to one. Doubling code is never a good idea